pkg/healthcheck/healthcheck.go

```diff
@@ -932,3 +960,39 @@ func validateDataPlanePodReporting(pods []*pb.Pod) error {
 
 	return nil
 }
+
+func checkUnschedulablePods(pods []corev1.Pod) error {
+	var podName []string
+	for _, pod := range pods {
+		for _, condition := range pod.Status.Conditions {
+			if condition.Reason == "Unschedulable" {
```

Comment:
Any reasons why we don't check the pod [phase](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase) instead? I'd think phases like [`Failed`, `Unknown`, `CrashLoopBackOff`](https://godoc.org/k8s.io/api/core/v1#PodPhase) etc. all imply that the pod isn't scheduled.

Reply:
i didn't knew that these all also imply that the pod isn't scheduled, if that is the case then probably we can check pod phase here itself, but i think it would be better to keep the two checks separate  i.e. the one occurring due to node failure and the others that might occur due to `Init:ImagePullBackOff` or `CrashLoopBackOff` , probably another check can be added  for the later

Reply:
also because pod phase only returns pending, running, failed or unknown we will have to check pod status and condition to know the actual reason of pod being unscheduled

Reply:
Yeah, checking the phase does feel a bit cumbersome. What do you think of just:

```
if condition.Type = corev1.PodReasonUnscheduleable
```

replacing the literal with the [`corev1.PodReasonUnscheduleable`](https://godoc.org/k8s.io/api/core/v1#PodConditionType) constant.

Reply:
yeah this can be done 👍